src/test/regress/expected/shared_connection_stats.out

```diff
@@ -495,7 +495,7 @@ BEGIN;
 (2 rows)
 
 ROLLBACK;
--- INSERT SELECT with RETURNING/ON CONFLICT clauses should honor shared_pool_size
+-- INSERT SELECT with RETURNING/ON CONFLICT clauses does not honor shared_pool_size
```

Comment:
This is a significant regression, but considering the correctness of the results and make this PR backportable, we go in this route.

Reply:
It's not too bad in terms of single node reliability, as we seem to be using "executing the copy locally for colocated file" in that case.

Reply:
Just for reference, that is implemented here: https://github.com/citusdata/citus/pull/4666